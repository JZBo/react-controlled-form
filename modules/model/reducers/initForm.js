/* @flow */
import type { Action } from '../../../types/Action'

export default function initForm(
  state: Object,
  { payload: { formId, initialFields = {}, initialState = {} } }: Action
): Object {
  return {
    ...state,
    [formId]: {
      data: initialFields,
      state: initialState,
    },
  }
}
