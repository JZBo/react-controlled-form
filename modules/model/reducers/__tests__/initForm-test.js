import initForm from '../initForm'

const payload = p => ({ payload: p })

describe('Initializing a form', () => {
  it('should add default data, state and isValid', () => {
    expect(initForm({}, payload({ formId: 'foo' }))).toMatchSnapshot()
  })

  it('should set initial data', () => {
    expect(
      initForm(
        {},
        payload({
          formId: 'foo',
          initialFields: {
            bar: {
              value: true,
              isRequired: false,
            },
            baz: {
              isEnabled: false,
            },
          },
        })
      )
    ).toMatchSnapshot()
  })

  it('should set an initial state', () => {
    expect(
      initForm(
        {},
        payload({
          formId: 'foo',
          initialState: {
            did_check: true,
          },
        })
      )
    ).toMatchSnapshot()
  })
})
