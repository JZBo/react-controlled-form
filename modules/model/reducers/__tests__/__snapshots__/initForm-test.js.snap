// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initializing a form should add default data, state and isValid 1`] = `
Object {
  "foo": Object {
    "data": Object {},
    "state": Object {
      "_initial": Object {},
    },
  },
}
`;

exports[`Initializing a form should set an initial state 1`] = `
Object {
  "foo": Object {
    "data": Object {},
    "state": Object {
      "_initial": Object {
        "did_check": true,
      },
      "did_check": true,
    },
  },
}
`;

exports[`Initializing a form should set initial data 1`] = `
Object {
  "foo": Object {
    "data": Object {
      "bar": Object {
        "_initial": Object {
          "isRequired": false,
          "value": true,
        },
        "isRequired": false,
        "value": true,
      },
      "baz": Object {
        "_initial": Object {
          "isEnabled": false,
        },
        "isEnabled": false,
      },
    },
    "state": Object {
      "_initial": Object {},
    },
  },
}
`;
